### ISSUE TITLE
Jython support

### ISSUE THREAD

**Issue by DRMacIver** [HypothesisWorks/hypothesis#174]
A couple of people have asked about Jython support and it would be nice to be able to offer it.

I'm currently aware of only a few major issues blocking it:
1. Jython is one of the platforms that don't support sqlite. See #173 for details.
2. Jython has a weird UCS-2 (it says UTF-16 but AIUI it's not) internal text encoding which means that for some codepoints unichr can throw a ValueError.
3. There are some bugs in py.test's pytester plugin under Jython
4. Jython triggers [my favourite pytz bug](https://bugs.launchpad.net/pytz/+bug/1435617) in an excitingly novel way.

Basically nothing serious, but it will require a lot of liasing with external projects and attention to detail to get everything working. It would be better if someone who were more interested in Jython than I am picked this up.

**Comment by Zac-HD** [comment]
Jython has been on life support since 2015, and seems more likely to die soon than support Python 3.  We'd still be happy to accept patches that improve Jython support, but with no comments here in three years we don't need a dedicated issue.
